### PR TITLE
Branch disjoint set

### PR DESCRIPTION
**Pull request by 4ndrelim**
Anyone pls help vet for correctness and clarity! (for DS portion can alr)

### CODE REVIEW

File: src/test/java/dataStructures/disjointSet/weightedUnion/DisjointSetTest.java
Comment (euchangxian) [inline src/test/java/dataStructures/disjointSet/weightedUnion/DisjointSetTest.java:1]: tests are well-constructed and easy to understand! 👍

File: src/test/java/dataStructures/disjointSet/quickFind/DisjointSetTest.java
Comment (euchangxian) [inline src/test/java/dataStructures/disjointSet/quickFind/DisjointSetTest.java:1]: 👍

File: src/main/java/dataStructures/disjointSet/weightedUnion/README.md [hunk 1]
```diff
@@ -30,9 +30,28 @@ are balanced.
...
+tree becomes a child of the larger tree.**
+It can be mathematically shown the height of the tree is bounded by O(log(n)).
+
+### Intuition - Why It Works
+First, it is crucial to know that Weighted Union's efficiency relies on careful **construction** of the trees. <br>
+Every element / object starts off in its own tree (i.e. its own component). When two components are merged, the smaller 
+objects of the smaller tree becomes part of the larger tree (by setting the root node of the smaller tree as a child).
+<br>Note that if the trees are of the same size, it does not matter which is assigned.
+
+Notice that trees will only increase in height when it's size is doubled. Working on this intuition, one can show
```
Comment (euchangxian) [inline src/main/java/dataStructures/disjointSet/weightedUnion/README.md:43]: nice

**Review by euchangxian — APPROVED** [review]
LGTM 👍 
I believe a possible enhancement to the READMEs would be including the practical applications of the data structures implemented. This would allow for better understanding of the problems that the DS solves!

**Comment by 4ndrelim** [comment]
do not merge yet - will add in applications before merging

**Pull request merged**